src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -66,12 +95,23 @@ internal void Walk(ISet<OpenApiTag>? tags)
             _visitor.Visit(tags);
 
             // Visit tags
-            if (tags != null)
+            if (tags is HashSet<OpenApiTag> { Count: 1 } hashSet && hashSet.First() is { } only)
             {
-                var tagsAsArray = tags.ToArray();
-                for (var i = 0; i < tagsAsArray.Length; i++)
+                WalkItem("0", only, static (self, item) => self.Walk(item));
```

Comment:
do you think that we could refactor this block and the next one into a generic method with a type constraint that derives from IOpenAPITag so we can avoid the duplication?